service/src/main/kotlin/fi/espoo/evaka/application/ApplicationControllerV2.kt

```diff
@@ -189,7 +189,7 @@ class ApplicationControllerV2(
 
         val userHasLimitedPrivileges = !user.hasOneOfRoles(Roles.ADMIN, Roles.FINANCE_ADMIN, Roles.SERVICE_WORKER)
         val authorizedUnitFilterList = when {
-            userHasLimitedPrivileges -> acl.getAuthorizedUnits(user).ids?.toList() ?: listOf()
+            userHasLimitedPrivileges -> acl.getAuthorizedUnits(user)
```

Comment:
You can even remove this `when` block completely :wink: 
`getAuthorizedUnits` returns either:

* `AclAuthorization.All`, which has null `.ids` property
* `AclAuthorization.Subset`, which has a non-null `.ids` property

So you can just pass a non-nullable `AclAuthorization`, which represents the idea of either allowing all or a subset of something

Reply:
Actually thought about this, but only later realised I also need to update the [null check](https://github.com/espoon-voltti/evaka/blob/ae73918c9fc3aaa8ccc4cda989943234c37ad9e5/service/src/main/kotlin/fi/espoo/evaka/application/ApplicationQueries.kt#L185) accordingly if the argument is non-nullable. 😄